File: benchmarks/programs/collatz/collatz.rb
total = 0
(1..10000).each do |i|
  x = i
  steps = 0
  while x != 1
    if x % 2 == 0
      x = x / 2
    else
      x = x * 3 + 1
    end
    steps += 1
  end
  total += steps
end
puts total

File: benchmarks/programs/collatz/collatz.js
let total = 0;
for (let i = 1; i <= 10000; i++) {
  let x = i;
  let steps = 0;
  while (x !== 1) {
    if (x % 2 === 0) {
      x = Math.floor(x / 2);
    } else {
      x = x * 3 + 1;
    }
    steps++;
  }
  total += steps;
}
console.log(total);
